### PR TITLE
adding oms-active-directory-security-audit-solution

### PR DESCRIPTION
**Pull request by AsithaD**
This template contain a OMS management solution -  Active Directory Security Audit Solution

### CODE REVIEW

**Comment by AsithaD** [comment]
create new resource group not supported with existing workspace

File: oms-active-directory-security-audit-solution/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+[![Deploy to Azure](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Foms-active-directory-security-audit-solution.json)
+
+This will send you to the Azure Portal with some default values for the template parameters. Ensure that the parameters reflects your setup so that you are deploying this into the existing Resource Group containing the Log Analytics Workspace and the Automation account.
+It is important that you type the exact values for your workspace name and resource group name, and points to the regions where these resources are deployed.
+
+![alt text](images/Template.png "Template")
+
+The ingestion will start 5-10 minutes post deployment.
+
+## Summery
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/README.md]: Summary

File: oms-active-directory-security-audit-solution/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+If you enabled auditing on domain controllers, it will generate a far amount of security data which you can use to search for events such as failed login or locked users. But searching this using native event viewer is not easy. OMS log analytics can be used to search these events and it will provide the information in rich presentable manner.
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/README.md]: fair amount...
using the native event...
information in a rich, presentable manner

File: oms-active-directory-security-audit-solution/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+This will send you to the Azure Portal with some default values for the template parameters. Ensure that the parameters reflects your setup so that you are deploying this into the existing Resource Group containing the Log Analytics Workspace and the Automation account.
+It is important that you type the exact values for your workspace name and resource group name, and points to the regions where these resources are deployed.
+
+![alt text](images/Template.png "Template")
+
+The ingestion will start 5-10 minutes post deployment.
+
+## Summery
+If you enabled auditing on domain controllers, it will generate a far amount of security data which you can use to search for events such as failed login or locked users. But searching this using native event viewer is not easy. OMS log analytics can be used to search these events and it will provide the information in rich presentable manner. 
+AD Security Audit solution is designed with number of these important use cases most customer face day in today's business. It will help you to identify
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/README.md]: customers

File: oms-active-directory-security-audit-solution/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+![alt text](images/Template.png "Template")
+
+The ingestion will start 5-10 minutes post deployment.
+
+## Summery
+If you enabled auditing on domain controllers, it will generate a far amount of security data which you can use to search for events such as failed login or locked users. But searching this using native event viewer is not easy. OMS log analytics can be used to search these events and it will provide the information in rich presentable manner. 
+AD Security Audit solution is designed with number of these important use cases most customer face day in today's business. It will help you to identify 
+*	User’s login to the domain with a time range
+*	Users failed to login to the domain, incorrect password attempts, number of attempts, time where the failed attempts are made.
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/README.md]: Users'...
time when...

File: oms-active-directory-security-audit-solution/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+## Deploying the Active Directory Security Audit Solution
+### Follow these instructions to deploy the solution into an existing OMS workspace
+Log into Azure Portal (https://portal.azure.com) and ensure you are in the subscription containing your OMS Workspace
+Locate your existing OMS Log Analytics Workspace and note the name of the workspace, the location of the workspace, and the Resource Group
+![alt text](images/WorkplaceInfo.png "Workspace Information")
+
+### You can now deploy the template
+[![Deploy to Azure](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Foms-active-directory-security-audit-solution.json)
+
+This will send you to the Azure Portal with some default values for the template parameters. Ensure that the parameters reflects your setup so that you are deploying this into the existing Resource Group containing the Log Analytics Workspace and the Automation account.
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/README.md]: reflect

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,768 @@
+{
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: format the JSON, e.g. SHIFT+ALT+F in VS Code

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,768 @@
...
+                "description": "Service Tier: Free, Standalone, or PerNode"
+            }
+        }
+      
+    },
+    
+
+
+    "variables": {
+        "omsSolutions": {
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove this it's never used

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,768 @@
...
+            "customSolution": {
+                "name": "AD Security Audit Solution",
+                "displayName": "AD Security Audit Solution",
+                "LogAnalyticsApiVersion": "2017-03-03-preview"
+            }
+        },
+        "LogAnalyticsApiVersion": "2017-03-03-preview",
+        "SolutionVersion": "1.1",
+        "SolutionName": "AD Security Audit Solution",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,768 @@
...
+
+    "variables": {
+        "omsSolutions": {
+            "customSolution": {
+                "name": "AD Security Audit Solution",
+                "displayName": "AD Security Audit Solution",
+                "LogAnalyticsApiVersion": "2017-03-03-preview"
+            }
+        },
+        "LogAnalyticsApiVersion": "2017-03-03-preview",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove this you'll need a literal apiVersion

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,768 @@
...
+        "SolutionVersion": "1.1",
+        "SolutionName": "AD Security Audit Solution",
+        "SolutionPublisher": "Asitha De Silva",
+        "ViewAuthor":  "asithahttp@gmail.com",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove/replace with an appropriate id here

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,768 @@
...
+        
+        },
+    
+
+"resources": [
+        {
+            "name": "[parameters('workspaceName')]",
+            "type": "Microsoft.OperationalInsights/workspaces",
+            "apiVersion": "2015-11-01-preview",
+            "location": "[parameters('workspaceRegionId')]",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,768 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+            "type": "string",
+          "metadata": {
+                "description": "Specify the workspace region"
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: update description

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 13]
```diff
@@ -0,0 +1,768 @@
...
+                "publisher": "[variables('SolutionPublisher')]",
+                "promotionCode": ""
+            }
+        },
+        {
+            "apiVersion": "[variables('LogAnalyticsApiVersion')]",
+            "name": "[concat(parameters('workspaceName'), '/', variables('ViewName'))]",
+            "type": "Microsoft.OperationalInsights/workspaces/views",
+            "location": "[parameters('workspaceregionId')]",
+            "id": "[Concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.OperationalInsights/workspaces/', parameters('workspaceName'),'/views/', variables('ViewName'))]",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove this

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,768 @@
...
+                                "Label": "Security Audit",
+                                
+                                
+                                "Link": {
+                                    "Label": "More info",
+                                    "Url": "https://asitha.info/"
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: use a valid url

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 15]
```diff
@@ -0,0 +1,768 @@
...
+                                }
+                            },
+                            "List": [
+                                {
+                                    "Title": "Active Directory Security Audit Solution",
+                                    "Content": "## Users and Groups Monitoring View\n1. Created user accounts\n2. Deleted user accounts\n3. Group membership changes - Domain Admin and Local Admin\n4. Related quries\n \n\nPost your feedback and Let us know how to improve this solution - **asithaed@millenniumitesp.com**,\n\n#### Solution by [Asitha De Silva](https://www.linkedin.com/in/asitha-de-silva-429b0927/ \"Asitha De Silva\")"
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: move personal contact info to readme.md

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 16]
```diff
@@ -0,0 +1,768 @@
...
+                            }
+                        }
+                    }
+                }
+            }
+        },
+
+        {
+                
+                    "apiVersion": "[variables('LogAnalyticsApiVersion')]",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: use a literal apiVersion

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 17]
```diff
@@ -0,0 +1,768 @@
...
+                    "name": "[concat(parameters('workspaceName'), '/', variables('ViewName2'))]",
+                    "type": "Microsoft.OperationalInsights/workspaces/views",
+                    "location": "[parameters('workspaceregionId')]",
+                    "id": "[Concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.OperationalInsights/workspaces/', parameters('workspaceName'),'/views/', variables('ViewName2'))]",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove this

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 18]
```diff
@@ -0,0 +1,768 @@
...
+                    "dependsOn": ["[concat('Microsoft.OperationalInsights/workspaces/', parameters('workspaceName'))]"
+                        
+                    ],
+                    "properties": {
+                        "Id": "[variables('ViewName2')]",
+                        "Name": "[variables('ViewName2')]",
+                        "DisplayName": "[variables('ViewName2')]",
+                        "Description": "[variables('ViewDescription')]",
+                        "Author": "asithahttp@gmail.com",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove/replace with an appropriate email addr

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 19]
```diff
@@ -0,0 +1,768 @@
...
...BCxKZqH6XXnrptE9khGzsK5yHmLGadrvGGmsUV199dcoXQHhIbptwkpEJ2DGMfJAPMjo2R3uNahMSEeVi1UuEg1i28eJxjpHWKEnk/PjHPz5jeVa8z6QjE3AOwGpnQIeETgJbotRFF12UhmMkHB0MCEY17AoY9hHOIxBPYvJ7++23T4tTAVN+LNESknKSkQk4B7AHDPkQhmYkx9lnn12svPLKiVAsLJBcEJClWZBTMiJWwaRjMcPPf/7zFO24h2N/RFsWQLjyZpKRCdgxXAnjggRA2yxWkV/96lfT0ns7F3ZE2KejwTpBe79UvwidmJtvvrkc5KbadQgG2OuedGQCdgyIhtAWVIhMUQBrA2nLrb766qlqNeIhdEiIeHRaWJJPBwbC2bvuJ5OOTMCOIQEjCSP5qI5jp4Hfl156aXHYYYcV2267bfHmN785LTT47Gc/m15WklSmN89eMunIBOwYkYCIxIjRD1Bl8psqlY6EY3hcQ6clDqkw1ufvSLY6mXRkAs4R6kiIEP2IenYgAMfspNjGAxyjsxHbejHfOpl0ZALOISIxYoTiNxEtdhxME8F5SIgYPWOedTLpyAScY1QJEiMfhLO6lXxUt0TE+YpMwDGgSkKPVSMeZIzHWNLFcI6RMrYLpxWZgGOC5KM6pU0H0SAW0S4Sy7YgIL3kYz9Gz2nF1BPQaS5guwhn4iCqLx1lw93fOg+He11cs0d6iEB6ZFiQd6xOBUSSaKSJZOI4adHB+yvkpXAd+XQp6sW96SzRO9dewGcirTaOz9mEeREBeWCqK6DhRHRiNBwOhHD0Mj3++9//Pm0F57wWGQboEu8LzAudfWEd4GyfA0jcSQTPhKDjHXfc8Z+j/12AG2d++mFeEJAohvMkG8AwHgeQ9Ec/+lFx4IEHFm94wxuK9ddfv3j1q19d7LTTTunYBRdckNLZzqqLPuY9KNBBMsUIwjH15H6QFWd+7WtfK971rncVG2+8cVrpjKyyyiqlvPSlLy3lZS97Waey5pprFi9+8YuLlVZaKX1ChC917b///sU555wzw948E/pDQITjbdqoU09AHevDAwyDU8WXvvSl9PK286hM7jvZz2/mWp/73OcmYvI9PgeBIaMkGYWE6MNCVK5FL3TlGGIB+dWvflUccMABSQ+m31h0wNQb87+KK2LiVB1zxF2K92HpF+sVWbHDb5aQrbv
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove personal information

File: oms-active-directory-security-audit-solution/azuredeploy.parameters.json [hunk 20]
```diff
@@ -0,0 +1,13 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+            "value": "GEN-UNIQUE-10"
+        },
+        
+        "workspaceRegionId": {
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.parameters.json]: remove this and use a defaultValue in the template

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 19 above)
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: remove personal info

Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: move to readme.md

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 16 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: u mean to remove the variable and add the apiversion here ?

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 17 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: please explain more what to remove from here

File: oms-active-directory-security-audit-solution/azuredeploy.parameters.json (hunk 20 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.parameters.json]: can you explain bit more on what to remove

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 19 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: done

**Comment by AsithaD** [comment]
most of the requested changes are fixed, i have added comments for others which i cannot understand to update. please refer the comments and reply, 
thanks

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 8 above)
Comment (bmoore-msft) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: this still needs to be removed

File: oms-active-directory-security-audit-solution/azuredeploy.json [hunk 21]
```diff
@@ -0,0 +1,743 @@
...
+            }
+        }
+    },
+    "variables": {
+       
+        "LogAnalyticsApiVersion": "2017-03-03-preview",
+        "SolutionVersion": "1.1",
+        "SolutionName": "AD Security Audit Solution",
+        "SolutionPublisher": "Asitha De Silva",
+        "ViewAuthor": "asithahttp@gmail.com",
```
Comment (bmoore-msft) [inline oms-active-directory-security-audit-solution/azuredeploy.json]: looks like many of these are not addressed, go back through the original comments and make sure you have each one covered

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 8 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: removed

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 9 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: done

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 10 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: added the ID

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 16 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: done

File: oms-active-directory-security-audit-solution/README.md [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+* #### OMS workplace name and Resource group name
+To deploy the solution you should know the OMS workspace name and the its related resource group name, for the current version you have to deploy it to the same resource group with the workspace
+
+## Deploying the Active Directory Security Audit Solution
+### Follow these instructions to deploy the solution into an existing OMS workspace
+Log into Azure Portal (https://portal.azure.com) and ensure you are in the subscription containing your OMS Workspace
+Locate your existing OMS Log Analytics Workspace and note the name of the workspace, the location of the workspace, and the Resource Group
+![alt text](images/WorkplaceInfo.png "Workspace Information")
+
+### You can now deploy the template
```
[the remark this answers is not in the document]
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/README.md]: removed

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 21 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: removed

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 17 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: i have two dashboards in my solution, i couldn't find a way to connect both views without this code. please give me a sample if you have, thanks

File: oms-active-directory-security-audit-solution/azuredeploy.json (hunk 13 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.json]: i have two dashboards in my solution, i couldn't find a way to connect both views without this code. please give me a sample if you have, thanks

**Comment by AsithaD** [comment]
hi bmoore-msft, your changes are addressed, please find the additional comments. 
Please check. 
thanks

File: oms-active-directory-security-audit-solution/azuredeploy.parameters.json (hunk 20 above)
Comment (AsithaD) [reply oms-active-directory-security-audit-solution/azuredeploy.parameters.json]: done

**Comment by AsithaD** [comment]
any updates on this 
thanks

**Pull request merged**